File: kotlin/src/main/kotlin/com/github/princesslana/smalld/examples/CatBot.kt
package com.github.princesslana.smalld.examples;

import java.net.URL
import com.github.kittinunf.fuel.*
import com.github.kittinunf.fuel.gson.*
import com.github.kittinunf.result.*

data class CatResponse(val file: String)

fun getCatUrl(): URL {
  return URL("http://aws.random.cat/meow".httpGet()
    .responseObject<CatResponse>()
    .third
    .getOrElse(CatResponse(""))
    .file)
}

val catBot = bot {
  fun onMessageCreate(msg: Message) {
    when (msg.content) {
      "++cat" -> post("/channels/${msg.channelId}/messages", Attachment("cat.jpg", "image/jpeg",getCatUrl()))
    }
  }

  onGatewayPayload { p ->
    when (p.t) {
      "MESSAGE_CREATE" -> onMessageCreate(gson.fromJson(p.d, Message::class.java))
    }
  }
}

fun main(args: Array<String>) = catBot.run(System.getenv("SMALLD_TOKEN"))

File: ruby/lib/catbot.rb

require_relative 'bot'

require 'httparty'
require 'json'

def send_cat(smalld, channel_id)
  response = HTTParty.get  'http://aws.random.cat/meow'

  cat_url = JSON.parse(response.body)['file']

  smalld.post(
    "/channels/#{channel_id}/messages",
    '',
    RAttachment.new('cat.jpg', 'image/jpeg', cat_url))
end

$cat_bot = Bot.new do |smalld|
  smalld.on_message_create do |msg|
    send_cat smalld, msg.channel_id if msg.content == '++cat'
  end
end

$cat_bot.run token: ENV['SMALLD_TOKEN'] if __FILE__ == $0
